service/src/main/java/uk/gov/hmcts/reform/fpl/service/children/ChildrenDataFixer.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.reform.fpl.service.children;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.fpl.utils.CaseDetailsHelper.isInOpenState;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDetailsHelper.isInReturnedState;
+
+@Component
+public class ChildrenDataFixer {
+
+    private final ObjectMapper mapper;
+
+    @Autowired
+    public ChildrenDataFixer(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public CaseDetails fix(final CaseDetails caseDetails) {
+        if (isInOpenState(caseDetails) || isInReturnedState(caseDetails)) {
+            return caseDetails;
+        }
+
+        Map<String, Object> data = caseDetails.getData();
+        final List<Element<Child>> children = mapper.convertValue(data.get("children1"), new TypeReference<>() {});
```

Comment:
maybe you dont even need ObjctMapper, you could do
((List)data.getOrDefault("children1", emptyList())).size()

or inject CaseConverter and convert entire caseDetails to caseData. It would be cleaner as you dont have to know that children is deserialised into 'children1'

Reply:
yeah, I kept on changing my mind on whether to do that or not

Reply:
it probs is cleaner, will update to use case converter